### ISSUE TITLE
Massive FPS drop when looking towards LP setup

### ISSUE THREAD

**Issue by ghost** [RS485/LogisticsPipes#1376]
Hello, one of the players on my server has a large LP setup, however it seems to be causing some massive FPS lag (server TPS seems unaffected). Standing at my base looking/walking away from his LP setup I get 100+ FPS, however as soon as I turn around and face the direction of his base (which is several hundred blocks away and just outside of my render distance), my FPS tanks to 20-25. Turning my back on his base brings my FPS back up to 100+. The other two people playing on the server have both reported the same. I was looking through the logs trying to find anything clues as to what was causing the massive drop, and came across this:

```
[22:42:22] [Chunk Batcher 7/ERROR] [CodeChickenLib]: 
 CCL has caught an exception whilst rendering a block
  BlockPos:      x:701, y:64, z:1027
  Block Class:   class logisticspipes.blocks.LogisticsSolidBlock
  Registry Name: logisticspipes:crafting_table
  Metadata:      0
  State:         logisticspipes:crafting_table[active=true,connection_0=true,connection_1=true,connection_2=true,connection_3=true,connection_4=true,connection_5=true,rotation=0]
 Tile at position
  Tile Class:    class logisticspipes.blocks.crafting.LogisticsCraftingTableTileEntity
  Tile Id:       logisticspipes:logistics_crafting_table
  Tile NBT:      {matrixitems:[],fuzzyFlags:[{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b}],rotation:0,x:701,matrixitemsCount:9,y:64,z:1027,placedBy_name:"",id:"logisticspipes:logistics_crafting_table",outputFuzzyFlags:{ignore_dmg:0b,ignore_nbt:0b,use_od:0b,use_category:0b},invitems:[],invitemsCount:18}
You can turn off player messages in the CCL config file.

net.minecraft.util.ReportedException: Tesselating block in world
	at net.minecraft.client.renderer.BlockRendererDispatcher.renderBlock(BlockRendererDispatcher.java:95) ~[bvm.class:?]
	at codechicken.lib.render.block.CCBlockRendererDispatcher.renderBlock(CCBlockRendererDispatcher.java:82) [CCBlockRendererDispatcher.class:?]
	at net.minecraft.client.renderer.chunk.RenderChunk.rebuildChunk(RenderChunk.java:200) [bxr.class:?]
	at net.minecraft.client.renderer.chunk.ChunkRenderWorker.processTask(SourceFile:100) [bxn.class:?]
	at net.minecraft.client.renderer.chunk.ChunkRenderWorker.run(SourceFile:43) [bxn.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
Caused by: java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909) ~[?:1.8.0_222]
	at java.util.ArrayList$Itr.next(ArrayList.java:859) ~[?:1.8.0_222]
	at net.minecraft.client.renderer.BlockModelRenderer.handler$beforeRenderModel$zbh000(BlockModelRenderer.java:555) ~[bvo.class:?]
	at net.minecraft.client.renderer.BlockModelRenderer.renderModel(BlockModelRenderer.java) ~[bvo.class:?]
	at net.minecraft.client.renderer.BlockModelRenderer.renderModel(BlockModelRenderer.java:38) ~[bvo.class:?]
	at net.minecraft.client.renderer.BlockRendererDispatcher.renderBlock(BlockRendererDispatcher.java:80) ~[bvm.class:?]
	... 5 more
[22:42:23] [Chunk Batcher 7/INFO] [net.minecraft.client.gui.GuiNewChat]: [CHAT] CCL Caught an exception rendering a block. See the log for info.
```

Which leads me to believe it may be LP related.

Details:
Minecraft 1.12.2
Logistics Pipes 0.10.2.206
Forge 2844

I can provide a full modlist if needed. Also am happy to provide any other additional information if needed.

**Comment by ghost** [comment]
After further testing and profiling with LagGoggles, it appears to specifically be caused by the players autocrafting setup. Here are some screenshots of the setup causing the lag:

![2019-09-12_21 23 00](https://user-images.githubusercontent.com/51542538/64868293-a5b7a500-d604-11e9-82ed-a25a4948f2d8.png)
![2019-09-12_21 23 32](https://user-images.githubusercontent.com/51542538/64868309-af410d00-d604-11e9-954e-f5836f9a94c1.png)
![2019-09-12_21 23 45](https://user-images.githubusercontent.com/51542538/64868323-b7994800-d604-11e9-94cc-dd788a3d3365.png)

**Comment by dblsaiko** [comment]
It's probably the crafting signs. The renderer on those is pretty inefficient. See #1315

**Comment by ghost** [comment]
Yup that seems to be it! I had the player remove all the crafting signs and the framerate no longer completely tanks when looking in the direction of this setup.

**Comment by bziemons** [comment]
Merged a first fix for 0.10.4.19
